### DISCUSSION TITLE
How to do a monthly or yearly analysis in RFM?

### DISCUSSION THREAD

**Discussion by franzoni315** [pymc-labs/pymc-marketing#1400] [Q&A]
In my company we sell yearly and monthly subscriptions with possibilities of cross-sell and upsell. How can I use the `rfm_summary` with monthly or yearly periods? When I try to set `time_unit="M"`, I get this error:

`ValueError: Unit M is not supported. Only unambiguous timedelta values durations are supported. Allowed units are 'W', 'D', 'h', 'm', 's', 'ms', 'us', 'ns'`

**Comment by ColtAllen** [comment]
Hey @franzoni315,

Use the `time_unit` and `time_scaler` parameters for this:

```python
rfm_summary(data,
            customer_id_col,
            datetime_col,
            time_unit="W", # "M" & "Y" are unsupported TimeDelta units in pandas,
            time_scaler=4.33, # so Weeks must be rescaled into Months
            )
```

**Comment by franzoni315** [comment]
It worked, thanks very much, @ColtAllen !

**Comment by franzoni315** [comment]
@ColtAllen I actually discovered that this parameter is not available for the `expected_customer_lifetime_value` function. I am trying to adapt the [CLV quickstart notebook](https://www.pymc-marketing.io/en/stable/notebooks/clv/clv_quickstart.html#clv-quickstart) for a yearly based business model, so for the rfm_summary I used the parameters `time_unit="W", time_scaler=55`, but I can't do the same for the `expected_customer_lifetime_value` function. I tried to set 

```
clv_estimate = gg.expected_customer_lifetime_value(
    transaction_model=bgm,
    data=rfm_data,
    future_t=55,  # year
    discount_rate=0.01,  # monthly discount rate ~ 12.7% annually
    time_unit="W")
```

but I am not sure if this is correct, and also it takes too long.

**Comment by ColtAllen** [comment]
@franzoni315 sorry for not seeing this sooner; I get a lot of emails related to `pymc-marketing` notifications.
> I actually discovered that this parameter is not available for the `expected_customer_lifetime_value` function.

It's not used in this function because `rfm_summary` aggregates data for model fitting, not predictions.

> ```
> clv_estimate = gg.expected_customer_lifetime_value(
>     transaction_model=bgm,
>     data=rfm_data,
>     future_t=55,  # year
>     discount_rate=0.01,  # monthly discount rate ~ 12.7% annually
>     time_unit="W")
> ```
> 
> but I am not sure if this is correct, and also it takes too long.

Set `time_unit='M'` and `future_t` to the number of months in the future you want predicted. "Years" is an unsupported time unit, but equivalent to `future_t=12*N_years`. This will also require the transaction model to be trained on monthly data.

Also, is this model only being used for cross-sells and upsells? If you're trying to predict if a customer will renew a subscription, `ShiftedBetaGeoModelIndividual` is the intended model.

However, for large customer populations `ShiftedBetaGeoModelIndividual` may take too long since it'll generate unique posteriors for each customer, and it can't currently be used with the `expected_customer_lifetime_value` function. This could be the proper motivation for us to create the population equivalent of this model and add discounted CLV support.
